interfaces/repo.go

```diff
@@ -267,10 +267,15 @@ func (r *Repository) RemoveSlot(snapName, slotName string) error {
 
 // Connect establishes a connection between a plug and a slot.
 // The plug and the slot must have the same interface.
-func (r *Repository) Connect(plugSnapName, plugName, slotSnapName, slotName string) error {
+func (r *Repository) Connect(ref *ConnRef) error {
```

Comment:
I was on the fence here but ultimately you are right. Since go's strings are immutable, with some optimisation this is proportional to 4 words. The cost should be irrelevant. Applied everywhere now.